scripts/leapp_preupgrade.py

```diff
@@ -0,0 +1,342 @@
+import json
+import os
+import subprocess
+
+
+JSON_REPORT_PATH = "/var/log/leapp/leapp-report.json"
+TXT_REPORT_PATH = "/var/log/leapp/leapp-report.txt"
+
+
+# Both classes taken from:
+# https://github.com/oamg/convert2rhel-worker-scripts/blob/main/scripts/preconversion_assessment_script.py
+class ProcessError(Exception):
+    """Custom exception to report errors during setup and run of leapp"""
+
+    def __init__(self, message):
+        super(ProcessError, self).__init__(message)
+        self.message = message
+
+
+class OutputCollector(object):
+    """Wrapper class for script expected stdout"""
+
+    def __init__(self, status="", message="", report="", entries=None):
+        self.status = status
+        self.message = message
+        self.report = report
+        self.tasks_format_version = "1.0"
+        self.tasks_format_id = "oamg-format"
+        self.entries = entries
+        self.report_json = None
+
+    def to_dict(self):
+        # If we have entries, then we change report_json to be a dictionary
+        # with the needed values, otherwise, we leave it as `None` to be
+        # transformed to `null` in json.
+        if self.entries:
+            self.report_json = {
+                "tasks_format_version": self.tasks_format_version,
+                "tasks_format_id": self.tasks_format_id,
+                "entries": self.entries,
+            }
+
+        return {
+            "status": self.status,
+            "message": self.message,
+            "report": self.report,
+            "report_json": self.report_json,
+        }
+
+
+def get_rhel_version():
+    """Currently we execute the task only for RHEL 7 or 8"""
+    print("Checking OS distribution and version ID ...")
+    try:
+        distribution_id = None
+        version_id = None
+        with open("/etc/os-release", "r") as os_release_file:
+            for line in os_release_file:
+                if line.startswith("ID="):
+                    distribution_id = line.split("=")[1].strip().strip('"')
+                elif line.startswith("VERSION_ID="):
+                    version_id = line.split("=")[1].strip().strip('"')
+            return distribution_id, version_id
+    except IOError:
+        print("Couldn't read /etc/os-release")
+    return None, None
```

Comment:
Since you're defining at the top that distribution_id and version_id are defined as None, we could refactor this a bit to use only one return statement in this function, like so:

```suggestion
    distribution_id = None
    version_id = None
    try:
        with open("/etc/os-release", "r") as os_release_file:
            for line in os_release_file:
                if line.startswith("ID="):
                    distribution_id = line.split("=")[1].strip().strip('"')
                elif line.startswith("VERSION_ID="):
                    version_id = line.split("=")[1].strip().strip('"')
    except IOError:
        print("Couldn't read /etc/os-release")
    return distribution_id, version_id
```

What do you think?

Reply:
sure makes sense, changed